boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala

```diff
@@ -117,7 +117,10 @@ object PicklerMaterializersImpl {
         q"""state.pickle[$fieldTpe](value.${accessor.name})"""
       }
 
-      q"""
+      if (sym.isDerivedValueClass)
```

Comment:
Why is this needed? We have [a check for this a few lines before](https://github.com/notxcain/boopickle/blob/d57ea13098e358d7f4a912e78cc20a0a627dec15/boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala#L95). How can we reach this point?

Reply:
That's strange indeed! I didn't notice this case.

Could you please take a look at https://github.com/notxcain/boopickle-repro, it fails to compile without this fix.

Reply:
That's strange but if I `publishLocal` from `master` my reproducer compiles. But it fails to compile with `1.2.6`

Reply:
Ah! This is not released yet in 1.2.6.

For now you can get it via jitpack:
```scala
resolvers += "jitpack" at "https://jitpack.io"
libraryDependencies += "com.github.suzaku-io.boopickle" %% "boopickle" % "master-SNAPSHOT"
```